pkg/client/linstor.go

```diff
@@ -990,3 +992,128 @@ func nil404(e error) error {
 	}
 	return e
 }
+
+func (s *Linstor) NodeExpand(source, target string) error {
+	fstype, err := GetDiskFormat(source)
+	if err != nil || fstype == "" {
+		return fmt.Errorf("error checking format for device %s: %v", source, err)
+	}
+
+	s.log.WithFields(logrus.Fields{
+		"device": fmt.Sprintf("%s", source),
+		"fstype": fmt.Sprintf("%s", fstype),
+	}).Info("expanding mounted volume")
+
+	// TODO: current only support exts3 ext4 xfs fs format.
+	switch fstype {
+	case "ext3", "ext4":
+		_, err = extResize(target)
+	case "xfs":
+		_, err = xfsResize(target)
+	default:
+		return fmt.Errorf("volume %s format is %s, unsupported expand currently",
+			source, fstype)
+	}
+	if err != nil {
+		return fmt.Errorf("resize filesystem failed, device: %s, path: %s, err: %v", source, target, err)
+	}
+
+	return nil
+}
```

Comment:
This and its helper functions should not be handrolled. Please use [existing k8s infrastructure](https://godoc.org/k8s.io/kubernetes/pkg/util/resizefs)